### PR TITLE
Add conversion self-service functionality

### PR DESCRIPTION
**Pull request by jeremy-cc**
Implements cancellation, date change and conversion split functions in public documented API

### CODE REVIEW

File: src/reference.yaml [hunk 1]
```diff
@@ -5984,7 +5984,7 @@ paths:
       x-api-group: convert
       summary: Get Conversion Cancellation quote
       description: Get a quote for the projected cost to cancel an existing conversion.
-      operationId: GetConversionCanceQuote
+      operationId: ConversionDateChangeQuote
```
Comment (WorkSutton) [inline src/reference.yaml]: Should this not be 'ConversionCancellationQuote'?

File: src/reference.yaml [hunk 2]
```diff
@@ -5736,6 +5736,970 @@ paths:
...
+              message: id should be in UUID format
+              params: ''
+            - code: on_behalf_of_self
+              category: on_behalf_of
+              message: You cannot act on behalf of your own Contact
+              params: ''
+            - code: contact_not_found
+              category: on_behalf_of
+              message: Contact was not found for this id
+              params: ''
```
Comment (WorkSutton) [inline src/reference.yaml]: Should we not also have:
- code: invalid_extra_parameters
              category: base
              message: Invalid extra parameters extra_parameter
              params: '{ "parameters" => "extra_parameter" }'

File: src/reference.yaml [hunk 3]
```diff
@@ -5736,6 +5736,970 @@ paths:
...
+          headers:
+            X-Request-Id:
+              type: integer
+              description: A unique reference for the request.
+  /conversions/{id}/date_change:
+    post:
+      tags:
+        - Conversions
+      x-api-group: convert
+      summary: Cancel Conversion
```
Comment (WorkSutton) [inline src/reference.yaml]: Should this be 'Conversion Date Change'?

File: src/reference.yaml [hunk 4]
```diff
@@ -5736,6 +5736,970 @@ paths:
...
+          required: true
+          type: string
+          description: ID of the conversion that is being changed
+        - name: on_behalf_of
+          in: formData
+          required: false
+          type: string
+          description: Contact UUID.
+        - name: new_settlement_date
+          required: false
```
Comment (WorkSutton) [inline src/reference.yaml]: Believe this should be `true` not `false`

File: src/reference.yaml [hunk 5]
```diff
@@ -5736,6 +5736,970 @@ paths:
...
+              message: id should be in UUID format
+              params: ''
+            - code: on_behalf_of_self
+              category: on_behalf_of
+              message: You cannot act on behalf of your own Contact
+              params: ''
+            - code: contact_not_found
+              category: on_behalf_of
+              message: Contact was not found for this id
+              params: ''
```
Comment (WorkSutton) [inline src/reference.yaml]: Believe we also want these additional errors:
```
            - code: invalid_extra_parameters
              category: base
              message: Invalid extra parameters extra_parameter
              params: '{ "parameters" => "extra_parameter" }'
            - code: new_settlement_date_invalid_in_past
              category: base
              message: New settlement date has to be today or greater than today's date
              params: ''
            - code: new_settlement_date_invalid
              category: base
              message: Invalid new settlement date, next valid date time is - YYYY-MM-DDT15:30:00Z
              params: ''
```

File: src/reference.yaml [hunk 6]
```diff
@@ -5736,6 +5736,970 @@ paths:
...
+          headers:
+            X-Request-Id:
+              type: integer
+              description: A unique reference for the request.
+  /conversions/{id}/split:
+    post:
+      tags:
+        - Conversions
+      x-api-group: convert
+      summary: Cancel Conversion
```
Comment (WorkSutton) [inline src/reference.yaml]: Should this not be `Conversion Split`

File: src/reference.yaml [hunk 7]
```diff
@@ -5736,6 +5736,970 @@ paths:
...
+              message: id should be in UUID format
+              params: ''
+            - code: on_behalf_of_self
+              category: on_behalf_of
+              message: You cannot act on behalf of your own Contact
+              params: ''
+            - code: contact_not_found
+              category: on_behalf_of
+              message: Contact was not found for this id
+              params: ''
```
Comment (WorkSutton) [inline src/reference.yaml]: Should we not include the following error?
```
            - code: invalid_extra_parameters
              category: base
              message: Invalid extra parameters extra_parameter
              params: '{ "parameters" => "extra_parameter" }'
```

File: src/reference.yaml (hunk 2 above)
Comment (derikvercueil) [reply src/reference.yaml]: Done

File: src/reference.yaml (hunk 5 above)
Comment (derikvercueil) [reply src/reference.yaml]: Done

File: src/reference.yaml (hunk 7 above)
Comment (derikvercueil) [reply src/reference.yaml]: Done

**Comment by derikvercueil** [comment]
/conversions/profit_and_loss needs to be added.

**Pull request merged**